pep-0581.rst

```diff
@@ -0,0 +1,310 @@
+PEP: 581
+Title: Using GitHub Issues for CPython
+Version: $Revision$
+Last-Modified: $Date$
+Author: Mariatta Wijaya <mariatta@python.org>
+Status: Draft
+Type: Process
+Content-Type: text/x-rst
+Created: 20-Jun-2018
+
+
+Abstract
+========
+
+This PEP outlines the steps required to migrate Python's issue tracker
+from Roundup to using GitHub Issues.
+
+
+Rationale
+=========
+
+CPython's development moved to GitHub on February 2017. All other projects within
+The PSF's organization are hosted on GitHub and are using GitHub issues.
+CPython is still using Roundup as the issue tracker in bugs.python.org (bpo).
+
+Why GitHub
+----------
+
+It has a lot of nice features readily available out of the box, that are not
+currently available in Roundup / bpo.
+
+- APIs we can use to build integrations and automations.
+
+- Ability to attach screenshots and files in pull requests and issues.
+
+- Administrators and core developers can edit issues comments and pull requests.
```

Comment:
I'm not sure that it's a big difference with bugs.python.org. The difference here is that the GitHub administrators can more easily give bug triage permission: single platform (GitHub) instead of two (GitHub + bugs.python.org). Single user directory.

Reply:
I cannot edit or delete a message in bpo.
If I made a mistake in my message, I had to post a new message to explain my previous mistake.
In GitHub I can edit my own messages, and core devs can edit everyone's messages.

Reply:
Github's all or nothing permission model is kind of a hindrance in this case. Right now there are triagers on bpo who aren't core devs, but can help categorize and resolve issues on bpo. Moving to Github means it would be harder to give them access.